### PR TITLE
v0.4.2

### LINKED ISSUE #6: Update lineage plots for agreements so that they are clearer to users
Plots should have agreements that replace/relate to each other connected to each other and amendments/protocols that modify a treaty connected to a specific treaty.

[https://app.zenhub.com/files/238751844/f6fc624c-5721-427c-959b-d08d56fb3312/download](https://app.zenhub.com/files/238751844/f6fc624c-5721-427c-959b-d08d56fb3312/download).

### LINKED ISSUE #39: `autographd()` arguments for node color and shape not working as expected for certain types of data/lists
DV <- lapply(ison_starwars, function(x) to_egos(x)$`DARTH VADER`)
DV <- DV[lengths(DV) != 0]
DV <- lapply(DV, function(x) 
  x %>% mutate(shape = ifelse(name == "DARTH VADER", "square", "circle"),
               color = ifelse(name == "DARTH VADER", "blue", "red")))
autographd(tlist = DV, node_size = 6, node_color = "color", node_shape = "shape")

The above does not shape or color nodes as expected.

### LINKED ISSUE #57: Please undo future::plan(...) changes
In:

https://github.com/stocnet/manynet/blob/8b1a435c91f3d930751ceccc2051f70a3869a846/R/make_play.R#L261

you're setting a new future strategy.  If you do that, it's important that you undo any changes, as explained and shown in Section 'For package developers' in [?future::plan](https://future.futureverse.org/reference/plan.html).

### PR DESCRIPTION
**Pull request by henriquesposito**
# Description

This PR fixes bugs with `autographr()` and `autographd()` functions.

## Making

- Closed #57 by updating `play_diffusions()` to revert future plan on exit
- Fixed bug with how `generate_random()` works for two-mode networks with specified number of ties

## Mapping

- Closed #6 by updating how "lineage" layout works and places nodes on Y axis
- Closed #39 by making `autographr()` more flexible and efficient in setting variables to aesthetics
- Updated themes to be compatible with newer and older versions of `{ggplot2}`
- Added "configuration" layout for small triad/quad networks

## Modifying

- Updated `to_reciprocated.matrix()` to consistently work with matrices

# Checklist:

- PR form
  - [x] Description above itemizes changes under subtitles, e.g. "## Data""
  - [x] Any closed, fixed, or related issues are referenced and explained in the description above, e.g. "Fixed #0 by adding A"
  - [x] Package builds on my OS without issues
- PR checks all pass for latest commit
  - [ ] CodeFactor check: Package improves or maintains good style
  - [x] Package builds on Mac
  - [x] Package builds on Windows
  - [x] Package builds on Linux
  - [ ] CodeCov check: Package improves or maintains good test coverage
- Documentation
  - [x] Any new or modified functions or data have roxygen style documentation in their .R scripts
  - [x] Longer functions are commented inline or broken down into helper functions so that it is easier to debug in the future
  - [x] PR description above and the NEWS.md file are aligned
  - [x] DESCRIPTION file version is bumped by the appropriate increment (major, minor, patch)

### CODE REVIEW

File: R/map_layout_partition.R [hunk 1]
```diff
@@ -239,52 +236,21 @@ layout_tbl_graph_lineage <- function(.data, rank, circular = FALSE) {
...
-      a[,1] <- seq(min(a[,1]), max(a[,1]), len = length(a[,1]))
-    }
-    adj <- rbind(adj, a)
-  }
-  name <- data.frame(names = out[,3])
-  out <- dplyr::left_join(name, adj, by = "names")
-  out <- apply(out[,2:3], 2, as.numeric)
-  rownames(out) <- name$names
-  out
-}
```
Comment (jhollway) [inline R/map_layout_partition.R:285]: No .adjust necessary?

File: R/map_layout_partition.R [hunk 2]
```diff
@@ -84,15 +84,13 @@ layout_tbl_graph_hierarchy <- function(.data, center = NULL,
...
       crd[which(is.nan(crd))] <- 0.5
-      crd[, 2] <- crd[, 2] * cos(pi) - crd[, 1] * sin(pi)
       rownames(crd) <- c(dimnames(net)[[1]], dimnames(net)[[2]])
     } else if (center == "events") {
       Act1 <- cbind(rep(0, ceiling(nrow(net)/2)), nrm(rng(ceiling(nn/2))))
       Act2 <- cbind(rep(2, floor(nrow(net)/2)), nrm(rng(floor(nn/2))))
       Evt <- cbind(rep(1, ncol(net)), nrm(rng(mm)))
       crd <- rbind(Act1, Act2, Evt)
       crd[which(is.nan(crd))] <- 0.5
-      crd[, 2] <- crd[, 2] * cos(pi) - crd[, 1] * sin(pi)
```
Comment (jhollway) [inline R/map_layout_partition.R:95]: Don't need to make rotated/circular anymore?

File: R/map_autographr.R [hunk 3]
```diff
@@ -55,7 +55,7 @@
 #'   Group variables should have a minimum of 3 nodes,
 #'   if less, number groups will be reduced by
 #'   merging categories with lower counts into one called "other".
-#' @param edge_color Tie variable to be used for coloring the nodes.
+#' @param edge_color Edge variable to be used for coloring the nodes.
```
Comment (jhollway) [inline R/map_autographr.R:58]: Please revert: while we use edge for graphical elements, users must refer to what is called a tie variable elsewhere in the package.

File: R/map_autographr.R [hunk 4]
```diff
@@ -1039,6 +1003,20 @@ map_dynamic <- function(edges_out, nodes_out, edge_color, node_shape,
   p
 }
 
+check_color <- function(v) {
```
Comment (jhollway) [inline R/map_autographr.R:1006]: New standard suggestion: functions that are only meant to be used internally should be prefixed with a period, e.g. `.check_color`

**Review by jhollway — APPROVED** [review]

File: R/map_layout_partition.R (hunk 2 above)
Comment (henriquesposito) [reply R/map_layout_partition.R:95]: Not anymore for this layout, I think this was changed in the latest version of `{ggraph}`.

**Pull request merged**